pkg/grpc/grpc.go

```diff
@@ -150,6 +152,13 @@ func NewGRPCServersFromConfigurationAndServe(configurations []*configuration.Ser
 		// Enable grpc reflection.
 		reflection.Register(s)
 
+		// Enable grpc health service
+		h := health.NewServer()
+		grpc_health_v1.RegisterHealthServer(s, h)
+		// TODO: construct an API for the caller to indicate when it is healthy and set this. Until then, we'll just track this function's extry and exit.
+		h.SetServingStatus(configuration.HealthCheckService, grpc_health_v1.HealthCheckResponse_SERVING)
+		defer h.SetServingStatus(configuration.HealthCheckService, grpc_health_v1.HealthCheckResponse_NOT_SERVING)
```

Comment:
What’s the purpose of this deferred statement? Don’t we already `log.Fatal()` in case gRPC has an error

Reply:
Just defensive, in case the other code ever changes.

Reply:
Please remove it for the time being then. This function isn't designed at all to allow continuation in case of failure. Adding `defer` statements like these may only cause us to incorrectly assume it is.

Reply:
Done